core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -31,25 +32,23 @@ import scala.collection.JavaConverters._
  * An implementation of Fiber that maintains context necessary for evaluation.
  */
 private[zio] final class FiberContext[E, A](
+  fiberId: FiberId,
+  @volatile var parentFiber: FiberContext[_, _],
   platform: Platform,
   startEnv: AnyRef,
   startExec: Executor,
   startIStatus: InterruptStatus,
-  startSStatus: SuperviseStatus,
+  startDStatus: Boolean,
   parentTrace: Option[ZTrace],
   initialTracingStatus: Boolean,
   fiberRefLocals: FiberRefLocals
-) extends Fiber[E, A] {
-  import java.util.{ Collections, Set }
+) extends Fiber[E, A] { self =>
 
   import FiberContext._
   import FiberState._
 
   // Accessed from multiple threads:
-  private[this] val state = new AtomicReference[FiberState[E, A]](FiberState.initial)
-
-  @noinline
-  private[this] def interrupted = state.get.interrupt
+  private val state = new AtomicReference[FiberState[E, A]](FiberState.initial)
```

Comment:
Not `private[this]`?

Reply:
I had to weaken it, because now I have to access the parent state from the child.